### PR TITLE
New Test Cases: Clip

### PR DESCRIPTION
**Pull request by Ac2zoom**
Hardtanh spec and backend test written.

### CODE REVIEW

**Comment by houseroad** [comment]
@Ac2zoom please rebase this PR to the master branch

**Comment by Ac2zoom** [comment]
Revising test cases for clip

**Comment by Ac2zoom** [comment]
@houseroad If I delete model.onnx, do I need to re-run generate-data?

**Comment by houseroad** [comment]
@onnxbot  retest this please

**Comment by houseroad** [comment]
Also `python gen_doc.py` to regenerate `Changelog.md` and `Operators.md`

File: onnx/backend/test/case/node/clip.py [hunk 1]
```diff
@@ -55,3 +77,23 @@ def export_clip_default():  # type: () -> None
...
+
+        x = np.array([-1, 0, 1]).astype(np.float32)
+        y = np.array([-1, 0, 1]).astype(np.float32)
+        expect(node, inputs=[x], outputs=[y],
+               name='test_clip_default_inbounds')
+
+        x = np.array([-1.1, 0, 1.1]).astype(np.float32)
+        y = np.array([-1, 0, 1]).astype(np.float32)
+        expect(node, inputs=[x], outputs=[y],
+               name='test_clip_default_outbounds')
```
Comment (houseroad) [inline onnx/backend/test/case/node/clip.py]: This case is incorrect

File: onnx/backend/test/case/node/clip.py [hunk 2]
```diff
@@ -55,3 +77,23 @@ def export_clip_default():  # type: () -> None
...
+
+        x = np.array([-1, 0, 1.1]).astype(np.float32)
+        y = np.array([-1, 0, 1]).astype(np.float32)
+        expect(node, inputs=[x], outputs=[y],
```
Comment (houseroad) [inline onnx/backend/test/case/node/clip.py]: This case is also incorrect.

File: onnx/backend/test/case/node/clip.py [hunk 3]
```diff
@@ -55,3 +77,23 @@ def export_clip_default():  # type: () -> None
         y = np.clip(x, -np.inf, 0.0)
         expect(node, inputs=[x], outputs=[y],
                name='test_clip_default_max')
+        node = onnx.helper.make_node(
+            'Clip',
+            inputs=['x'],
+            outputs=['y'],
+        )
+
+        x = np.array([-1, 0, 1]).astype(np.float32)
```
Comment (houseroad) [inline onnx/backend/test/case/node/clip.py]: Just keep this one.

File: onnx/backend/test/case/node/clip.py [hunk 4]
```diff
@@ -55,3 +77,23 @@ def export_clip_default():  # type: () -> None
...
+        y = np.array([-1, 0, 1]).astype(np.float32)
+        expect(node, inputs=[x], outputs=[y],
+               name='test_clip_default_inbounds')
+
+        x = np.array([-1.1, 0, 1.1]).astype(np.float32)
+        y = np.array([-1.1, 0, 1.1]).astype(np.float32)
+        expect(node, inputs=[x], outputs=[y],
```
Comment (houseroad) [inline onnx/backend/test/case/node/clip.py]: remove this one, it's the same as the previous one, don't forget to delete the corresponding binary data. :-)

File: onnx/backend/test/case/node/clip.py [hunk 5]
```diff
@@ -55,3 +77,23 @@ def export_clip_default():  # type: () -> None
...
+               name='test_clip_default_outbounds')
+
+        x = np.array([-1, 0, 1.1]).astype(np.float32)
+        y = np.array([-1, 0, 1.1]).astype(np.float32)
+        expect(node, inputs=[x], outputs=[y],
```
Comment (houseroad) [inline onnx/backend/test/case/node/clip.py]: Please also remove this one, it's the same as the previous one, don't forget to delete the corresponding binary data. :-)

**Comment by houseroad** [comment]
@Ac2zoom regenerate the doc please :-)

**Pull request merged**